lib/daemon_runner/semaphore.rb

```diff
@@ -0,0 +1,152 @@
+module DaemonRunner
+  #
+  # Manage semaphore locks with Consul
+  #
+  class Semaphore
+    include Logger
+
+    class << self
+
+      # Create a session
+      #
+      # @param name [String] The name of the session, usually the service name
+      # @see #initialize for extra `options`
+      def start(name, **options)
+        options.merge!(name: name)
+        @lock ||= Semaphore.new(options)
+      end
+
+      # Acquire a lock with the current session
+      #
+      # @param limit [Integer] The number of nodes that can request the lock
+      # @return [Boolean] `true` if the lock was acquired
+      #
+      def lock(limit = 3)
+        raise RuntimeError 'Must call start first' if @lock.nil?
+        @lock.limit = limit
+        @lock.contender_key
+        @lock.semaphore_state
```

Comment:
We need to check that the "Limit" in the semaphore state matches the `limit` provided as an argument. If they're not the same, we shouldn't try to write the lock.